src/ngx_http_lua_headers_in.c

```diff
@@ -645,14 +645,95 @@ ngx_http_clear_builtin_header(ngx_http_request_t *r,
 }
 
 
+static ngx_hash_init_t  lua_req_header_hash;
+static u_char           header_string[512];
```

Comment:
header_string does not need to be global either, as far as I can tell, this array is used to clone "keys" which will be eventually fed to  ngx_hash_init(). In ngx_hash_init(), keys are cloned again, making this variable dead.

Using global variable is in general bad, we should avoid whenever possible. There is another reason for avoid using global variables in lua-nginx-module: currently, at least on Linux, luajit can allocate memory from the (the-end-of-.bss, 2G], space, and this variable make .bss 512 byte bigger.